src/sonic-config-engine/minigraph.py

```diff
@@ -2068,6 +2140,18 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
             'client_crt_cname': 'client.restapi.sonic'
         }
     }
+    if devices[hostname]["type"] == "BmcMgmtToRRouter":
+        dhcp_server_ipv4, dhcp_server_ipv4_port, dhcp_server_ipv4_customized_options = \
+            generate_ipv4_dhcp_server_config(devices, link_ports, port_alias_map, vlan_intfs, vlan_members,
```

Comment:
>generate_ipv4_dhcp_server_config [](http://example.com/codeflow?start=12&length=32)

Why generate_ipv4_dhcp_server_config() is needed. It just manipulated existing ConfigDB, does not parse minigraph at all, no extra info added into ConfigDB. Seems like everything you needed is already in ConfigDB or as default values.